src/yuzu/main.cpp

```diff
@@ -2587,13 +2587,48 @@ void GMainWindow::OnConfigure() {
             &GMainWindow::OnLanguageChanged);
 
     const auto result = configure_dialog.exec();
-    if (result != QDialog::Accepted) {
+    if (result != QDialog::Accepted && !UISettings::values.reset_to_defaults) {
         return;
     }
 
-    configure_dialog.ApplyConfiguration();
+    if (UISettings::values.reset_to_defaults) {
+        LOG_INFO(Frontend, "Resetting all settings to defaults");
+        if (!Common::FS::RemoveFile(config->GetConfigFilePath())) {
+            LOG_WARNING(Frontend, "Failed to remove configuration file");
+        }
+        if (!Common::FS::RemoveDirContentsRecursively(
+            Common::FS::GetYuzuPath(Common::FS::YuzuPath::ConfigDir) / "custom")) {
+            LOG_WARNING(Frontend, "Failed to remove custom configuration files");
+        }
+        if (!Common::FS::RemoveDirRecursively(
+                   Common::FS::GetYuzuPath(Common::FS::YuzuPath::CacheDir) / "game_list")) {
+            LOG_WARNING(Frontend, "Failed to remove game metadata cache files");
+        }
+
+        // Explicitly save the game directories, since reinitializing config does not do so.
+        QVector<UISettings::GameDir> old_game_dirs = UISettings::values.game_dirs;
+        QVector<u64> old_favorited_ids = UISettings::values.favorited_ids;
+
+        Settings::values.disabled_addons.clear();
+
+        config = std::make_unique<Config>();
+        UISettings::values.reset_to_defaults = false;
+
+        UISettings::values.game_dirs = old_game_dirs;
```

Comment:
```suggestion
        UISettings::values.game_dirs = std::move(old_game_dirs);
```

Ditto for `old_favorited_ids`.